kernel/src/device/pty/master.rs

```diff
@@ -176,11 +176,10 @@ impl FileIo for PtyMaster {
 
 impl Drop for PtyMaster {
     fn drop(&mut self) {
-        let fs = self.ptmx.fs();
-        let devpts = fs.downcast_ref::<DevPts>().unwrap();
-
-        let index = self.slave.index();
-        devpts.remove_slave(index);
+        if let Some(devpts) = self.ptmx.devpts() {
```

Comment:
> The third commit fixes another issue: when running Podman, PtyMaster's Drop implementation could panic because devpts had already been dropped. I couldn't determine the root cause, so as a workaround the code now skips the unwrapping step when devpts has been dropped.

Unless someone unmounts `/dev/pts`... I don't understand why this can happen?

Reply:
It sounds reasonable. Then this seems to be an issue that affects other file systems/inodes as well?